src/ess/reduce/nexus.py

```diff
@@ -0,0 +1,345 @@
+    *,
+    source_name: Optional[SourceName] = None,
+    entry_name: Optional[EntryName] = None,
+) -> RawSource:
+    """Load a source from a NeXus file.
+
+    The source position is computed automatically for NeXus transformations.
+
+    Parameters
+    ----------
+    file_path:
+        Indicates where to load data from.
+        One of:
+
+        - Path to a NeXus file on disk.
+        - File handle or buffer for reading binary data.
+        - A ScippNexus group of the root of a NeXus file.
+    source_name:
+        Name of the source to load.
+        Must be a group in an instrument group in the entry (see below).
+        If ``None``, the source will be located based
+        on its NeXus class.
+    entry_name:
+        Name of the instrument that contains the source.
+        If ``None``, the entry will be located based
+        on its NeXus class, but there cannot be more than 1.
+
+    Returns
+    -------
+    :
+        A data group containing all data stored in
+        the source NeXus group.
+    """
+    return RawSource(
+        _load_group_with_positions(
+            file_path,
+            group_name=source_name,
+            nx_class=snx.NXsource,
+            entry_name=entry_name,
+        )
+    )
+
+
+def load_sample(
+    file_path: Union[FilePath, NeXusFile, NeXusGroup],
+    entry_name: Optional[EntryName] = None,
+) -> RawSample:
+    """Load a sample from a NeXus file.
+
+    The sample is located based on its NeXus class.
+    There can be only one sample in a NeXus file or
+    in the entry indicated by ``entry_name``.
+
+    Parameters
+    ----------
+    file_path:
+        Indicates where to load data from.
+        One of:
+
+        - Path to a NeXus file on disk.
+        - File handle or buffer for reading binary data.
+        - A ScippNexus group of the root of a NeXus file.
+    entry_name:
+        Name of the instrument that contains the source.
+        If ``None``, the entry will be located based
+        on its NeXus class, but there cannot be more than 1.
+
+    Returns
+    -------
+    :
+        A data group containing all data stored in
+        the sample NeXus group.
+    """
+    with _open_nexus_file(file_path) as f:
+        entry = _unique_child_group(f, snx.NXentry, entry_name)
+        loaded = cast(sc.DataGroup, _unique_child_group(entry, snx.NXsample, None)[()])
+    return RawSample(loaded)
+
+
+def _load_group_with_positions(
+    file_path: Union[FilePath, NeXusFile, NeXusGroup],
+    *,
+    group_name: Optional[str],
+    nx_class: Type[snx.NXobject],
+    entry_name: Optional[EntryName] = None,
+) -> sc.DataGroup:
+    with _open_nexus_file(file_path) as f:
+        entry = _unique_child_group(f, snx.NXentry, entry_name)
+        instrument = _unique_child_group(entry, snx.NXinstrument, None)
+        loaded = cast(
+            sc.DataGroup, _unique_child_group(instrument, nx_class, group_name)[()]
+        )
+        loaded = snx.compute_positions(loaded)
```

Comment:
We have to support the `store_transform` arg. The transforms are required in SANS (for the detectors). Probably for now we can just always enable this, we can revisit later if it turns out there are issues with it.

Reply:
They are stored in the data group. Does the sans code need to change the name of the item?

Reply:
We already agreed to just store it as "transform".